v03_pipeline/lib/methods/relatedness.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import hail as hl
+from gnomad.sample_qc.pipeline import filter_rows_for_qc
+
+from v03_pipeline.lib.model import ReferenceGenome
+
+
+# Stub port from https://github.com/broadinstitute/gnomad_qc/blob/9763d8a581665ed0be8bf2ea09d74e3cabc4806e/gnomad_qc/v2/resources/sample_qc.py#L40
+def qc_mt_path(reference_genome: ReferenceGenome) -> str:
+    ref_str = '.grch38' if reference_genome == ReferenceGenome.GRCh38 else ''
+    return f'gs://gnomad/sample_qc/mt/gnomad.joint.high_callrate_common_biallelic_snps.pruned{ref_str}.mt'
+
+
+# Stub port from https://github.com/broadinstitute/gnomad_qc/blame/9763d8a581665ed0be8bf2ea09d74e3cabc4806e/gnomad_qc/v3/resources/sample_qc.py#L219
+def qc_path() -> str:
+    return 'gs://gnomad/sample_qc/mt/genomes_v3.1/gnomad_v3.1_qc_mt_v2_sites_dense.mt'
+
+
+def filter_and_ld_prune(
+    mt: hl.MatrixTable,
+    reference_genome: ReferenceGenome,
+    use_gnomad_in_ld_prune: bool,
+) -> hl.MatrixTable:
+    mt = filter_rows_for_qc(
+        mt,
+        min_af=0.001,
+        min_callrate=0.99,
+        apply_hard_filters=False,
+    )
+    if not use_gnomad_in_ld_prune:
+        mm_pruned = hl.ld_prune(mt.GT, r2=0.1)
+        return mt.filter_rows(hl.is_defined(mm_pruned[mt.row_key]))
+    if reference_genome == ReferenceGenome.GRCh37:
+        pruned_mt = hl.read_matrix_table(qc_mt_path(reference_genome))
+    elif reference_genome == ReferenceGenome.GRCh38:
+        pruned_mt = hl.read_matrix_table(qc_path())
```

Comment:
wouldn't is be neater to just to modify `qc_mt_path(reference_genome)` to work properly for both 37 and 38? This is the only place it is used and you would expect the helper function that takes a reference_genome argument to work with multiple reference_genomes and not just one

Reply:
Yeah, I left the code structure as is was in `ibd.py`, but might as well change it to make it make more sense.